src/System.Resources.Reader/src/System/Resources/StreamExtensions.cs

```diff
@@ -78,13 +80,35 @@ public static string ReadString(this Stream stream)
                 int justRead = stream.Read(buffer, totalRead, stringLength - totalRead);
                 if(justRead == 0)
                 {
-                    throw new BadImageFormatException(SR.Resources_StreamNotValid);
+                    throw new EndOfStreamException(SR.BadImageFormat_ResourceNameCorrupted_NameIndex + position);
```

Comment:
I think this needs to be done with SR.Format, with the resource string defined to take a hole that's filled in by position.